### ISSUE TITLE
Windows: Error casting column if empty in virtual_table.cpp

### ISSUE THREAD

**Issue by stachdude** [facebook/osquery#2985]
Hi again,

Same build https://github.com/facebook/osquery/commit/78ed32a67346d67fd0ecd96c4353ffe666caa8bd I know there is a newer build, but I think this hasn't been changed in newer commits.

this time I'm testing `--disable_database=true` on Windows2008R2 and stumble on situation where if result of column is empty we get an casting error.

Here is query
```
b8d4af7: SELECT DISTINCT processes.name, processes.path, processes.cmdline, proc
esses.cwd, processes.root, processes.threads, listening_ports.address, listening
_ports.protocol, listening_ports.port FROM listening_ports JOIN processes USING
(pid) WHERE listening_ports.address NOT IN ('127.0.0.1', '::') and processes.pat
h != '';
```

and log from console

```
I0208 12:35:19.390625  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.421875  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.468750  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.484375  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.515625  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.546875  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.562500  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.625000  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.656250  3068 virtual_table.cpp:264] Error casting threads () to I
NTEGER
I0208 12:35:19.781250  3068 scheduler.cpp:129] Found results for query: test_3_9
b8d4af7
```

And here is JSON result

```
{'action': 'added',
 'calendarTime': 'Wed Feb  8 20:09:02 2017 UTC',
 'columns': {'address': '0.0.0.0',
             'cmdline': 'wininit.exe',
             'cwd': 'C',
             'name': 'wininit.exe',
             'path': 'C:\\Windows\\system32\\wininit.exe',
             'port': '49152',
             'protocol': '6',
             'root': 'C',
             'threads': ''},
 'hostIdentifier': 'WIN-K5KF2SDPS87',
 'name': 'test_3_9b8d4af7',
 'unixTime': '1486584542'}
```

**Comment by stachdude** [comment]
Ok, I Just realized - of course after I posted this issue - this is an Information error.

**Comment by poppyseedplehzr** [comment]
Hah, thanks @stachdude, we definitely need an issue for this anyway. Iirc this error message is coming from the processes table which isn't setting the `threads` field to a value, and thus causing the casting issue. It should be a relatively easy fix but didn't seem to be hurting anyone. I'll see if I can get something up here shortly :)

**Comment by vbisserie** [comment]
Hi,
I observed the same behavior on OSX with osquery version [d0945b0](https://github.com/facebook/osquery/commit/d0945b0f4e4600893fd32d0f6167554363edd56c).

Log : 
`virtual_table.cpp:291] Error casting time () to INTEGER`

Query:
`SELECT * FROM shell_history;`

**Comment by theopolis** [comment]
The most common cause for these verbose messages (shown when you use `--verbose`) is the table author not filling in a value for the column. These are nice to have in verbose to remind folks to go back and 'fix' the table. :)

**Comment by muffins** [comment]
Looks like this specific issues has been resolved with #3536. Feel free to open another issue if this problem persists!
